client/dashboard/me/tax-details/index.tsx

```diff
@@ -1,11 +1,167 @@
+import { useResetSupportInteraction } from '@automattic/help-center/src/hooks/use-reset-support-interaction';
+import {
+	Card,
+	CardBody,
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { useDispatch as useDataStoreDispatch, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback } from 'react';
+import { useAnalytics } from '../../app/analytics';
+import { useTaxName } from '../../app/hooks/use-country-list';
+import { useGeoLocationQuery } from '../../app/queries/geolocation';
+import InlineSupportLink from '../../components/inline-support-link';
 import { PageHeader } from '../../components/page-header';
 import PageLayout from '../../components/page-layout';
+import { localizeUrl } from './localize-url';
+import useRecordUserTaxEvents from './use-record-user-tax-events';
+import useUserTaxDetails from './use-user-tax-details';
+import UserTaxForm from './user-tax-form';
+import './style.scss';
+
+const HELP_CENTER_STORE = 'automattic/help-center';
+
+export default function UserTaxInfoPage() {
+	const { recordTracksEvent } = useAnalytics();
+	const translate = useTranslate();
+	const { data: geoData } = useGeoLocationQuery();
+	const { fetchError, userTaxDetails } = useUserTaxDetails();
+	const taxName = useTaxName(
+		userTaxDetails.country ?? userTaxDetails.country ?? geoData?.country_short ?? 'GB'
+	);
+	const resetSupportInteraction = useResetSupportInteraction();
+
+	const { setShowHelpCenter, setNavigateToRoute } = useDataStoreDispatch( HELP_CENTER_STORE );
+
+	const reduxDispatch = useDispatch();
+
+	/* This is a call to action for contacting support */
+	const contactSupportLinkTitle = translate( 'Contact Happiness Engineers' );
+
+	const taxSupportPageURL = localizeUrl( 'https://wordpress.com/support/vat-gst-other-taxes/' );
+
+	/* This is the title of the support page from https://wordpress.com/support/vat-gst-other-taxes/ */
+	const taxSupportPageLinkTitle = translate( 'VAT, GST, and other taxes' );
+
+	const handleOpenCenterChat = useCallback(
+		async ( e: React.MouseEvent< HTMLAnchorElement > ) => {
+			e.preventDefault();
+			setNavigateToRoute( '/odie' );
+			setShowHelpCenter( true );
+			await resetSupportInteraction();
+			reduxDispatch( recordTracksEvent( 'calypso_vat_details_support_click' ) );
+		},
+		[
+			reduxDispatch,
+			recordTracksEvent,
+			resetSupportInteraction,
+			setNavigateToRoute,
+			setShowHelpCenter,
+		]
+	);
+
+	useRecordUserTaxEvents( { fetchError } );
+
+	if ( fetchError ) {
```

Comment:
By fetching the tax details in the router you will need to handle tax specific error UI in the router too. E.g. tell it which error component to display. Otherwise you will get our generic error handler.